parser/utils.ts

```diff
@@ -0,0 +1,25 @@
+const htmlUnescapes = {
+  '&amp;': '&',
+  '&lt;': '<',
+  '&gt;': '>',
+  '&quot;': '"',
+  '&#x27;': "'",
+  '&#x60;': '`',
+  '&#32': ' ',
+};
+
+const reEscapedHtml = /&(?:amp|lt|gt|quot|#(x27|x60|32));/g;
+const reHasEscapedHtml = RegExp(reEscapedHtml.source);
+
+/**
+ * The inverse of `escape`this method converts the HTML entities
+ * `&amp;`, `&lt;`, `&gt;`, `&quot;` and `&#39;` in `string` to
+ * their corresponding characters.
+ * Source: https://github.com/lodash/lodash/blob/main/src/unescape.ts
+ */
+function unescape(string: string): string {
```

Comment:
There's already a function called `unescape` in JS ([link](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/unescape)), so let's rename it to `unescapeText` to avoid confusion.